controllers/cache/utils.go

```diff
@@ -0,0 +1,214 @@
+package cache
+
+import (
+	"bytes"
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	labels1 "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"math/rand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+)
+
+const (
+	ReidsType    = "rfr"
+	SentinelType = "rfs"
+
+	RoleName          = "harbor-cluster"
+	RedisSentinelPort = "26379"
+	redisRoleMaster   = "role:master"
+)
+
+// GetRedisName returns the name for redis resources
+func (redis *RedisReconciler) GetRedisName() string {
+	return generateName(ReidsType, redis.GetHarborClusterName())
+}
+
+func generateName(typeName, metaName string) string {
+	return fmt.Sprintf("%s-%s", typeName, metaName)
+}
+
+func RandomString(randLength int, randType string) (result string) {
```

Comment:
If other controllers also needs such a name generator, we should consider moving it to the shared lib package.